src/Qbead.h

```diff
@@ -103,6 +83,133 @@ void connect_callback(uint16_t conn_handle)
   Serial.println(central_name);
 }
 
+// State Class Definition
+class State {
+private:
+    float x, y, z;    // Cartesian coordinates
+    float theta, phi; // Spherical coordinates
+
+    void cartesianToSpherical();
+    void sphericalToCartesian();
+
+public:
+    State();
+    State(float x_init, float y_init, float z_init);
+    State(float theta_init, float phi_init);
+
+    void setX(float new_x);
+    void setY(float new_y);
+    void setZ(float new_z);
+    void setXYZ(float new_x, float new_y, float new_z);
+
+    void setTheta(float new_theta);
+    void setPhi(float new_phi);
+    void setThetaPhi(float new_theta, float new_phi);
+
+    float getX() const;
+    float getY() const;
+    float getZ() const;
+
+    float getTheta() const;
+    float getPhi() const;
+
+    void printState();
+};
+
+State::State() : x(0), y(0), z(1) {
+    cartesianToSpherical();
+}
+
+State::State(float x_init, float y_init, float z_init) : x(x_init), y(y_init), z(z_init) {
+    cartesianToSpherical();
+}
+
+State::State(float theta_init, float phi_init) : theta(theta_init), phi(phi_init) {
+    sphericalToCartesian();
+}
+
+void State::cartesianToSpherical() {
+    float l = sqrt(x * x + y * y + z * z);
+    if (l == 0) l = 1;
+    theta = acos(z / l) * 180 / PI;
+    phi = atan2(y, x) * 180 / PI;
+    if (phi < 0) phi += 360;
+}
+
+void State::sphericalToCartesian() {
+    float r = 1;
+    x = r * sin(theta * PI / 180.0) * cos(phi * PI / 180.0);
+    y = r * sin(theta * PI / 180.0) * sin(phi * PI / 180.0);
+    z = r * cos(theta * PI / 180.0);
+}
+
+void State::setX(float new_x) {
+    x = new_x;
+    cartesianToSpherical();
+}
+
+void State::setY(float new_y) {
+    y = new_y;
+    cartesianToSpherical();
+}
+
+void State::setZ(float new_z) {
+    z = new_z;
+    cartesianToSpherical();
+}
+
+void State::setXYZ(float new_x, float new_y, float new_z) {
+    x = new_x;
+    y = new_y;
+    z = new_z;
+    cartesianToSpherical();
+}
+
+void State::setTheta(float new_theta) {
+    theta = new_theta;
+    sphericalToCartesian();
+}
+
+void State::setPhi(float new_phi) {
+    phi = new_phi;
+    sphericalToCartesian();
+}
+
+void State::setThetaPhi(float new_theta, float new_phi) {
+    theta = new_theta;
+    phi = new_phi;
+    sphericalToCartesian();
+}
+
+float State::getX() const {
+    return x;
+}
+
+float State::getY() const {
+    return y;
+}
+
+float State::getZ() const {
+    return z;
+}
+
+float State::getTheta() const {
+    return theta;
+}
+
+float State::getPhi() const {
+    return phi;
+}
+
+void State::printState() {
+    Serial.print("Cartesian: x = "); Serial.print(x);
+    Serial.print(", y = "); Serial.print(y);
+    Serial.print(", z = "); Serial.println(z);
+
+    Serial.print("Spherical: theta = "); Serial.print(theta);
+    Serial.print(", phi = "); Serial.println(phi);
+}
```

Comment:
Could you move these inside of the class declaration? No need to have them repeated like this if we have a single header file.